File: client/src/app/_models/admin/customer.ts
import { IAgencySpecialty } from "./agencySpecialty";
import { ICustomerIndustry } from "./customerIndustry";
import { ICustomerOfficial } from "./customerOfficial";
import { ICustomerOfficialDto } from "./customerOfficialDto";
import { IVendorSpecialty } from "./vendorSpecialty";


export interface ICustomer {
     id: number;
     customerType: string;
     customerName: string;
     knownAs: string;
     add: string;
     add2: string;
     city: string;
     pin: string;
     district: string;
     state: string;
     country: string;
     email: string;
     website: string;
     phone: string;
     phone2: string;
     logoUrl?: string;
     customerStatus: number;
     createdOn: Date;
     introduction: string;
     customerIndustries: ICustomerIndustry[];
     customerOfficials: ICustomerOfficial[];
     agencySpecialties: IAgencySpecialty[];
     vendorSpecialties: IVendorSpecialty[];
     lastActive: Date;
 }

 export class Customer implements ICustomer {
    id = 0;
    customerType= '';
    customerName= '';
    knownAs= '';
    add= '';
    add2= '';
    city= '';
    pin= '';
    district= '';
    state= '';
    country= '';
    email= '';
    website= '';
    phone= '';
    phone2= '';
    logoUrl?= '';
    customerStatus= 0;
    createdOn = new Date('1900-01-01');

    introduction= '';
    customerIndustries: ICustomerIndustry[]=[];
    customerOfficials: ICustomerOfficial[]=[];
    agencySpecialties: IAgencySpecialty[]=[];
    vendorSpecialties: IVendorSpecialty[]=[];
    
    lastActive = new Date('1900-01-01');
 }

File: api/Entities/Admin/Client/Customer.cs
using System.ComponentModel.DataAnnotations;

namespace api.Entities.Admin.Client
{
    public class Customer: BaseEntity
    {
        [Required]
        public string CustomerType { get; set; }
        [Required, MaxLength(100), MinLength(5)]
        public string CustomerName { get; set; }
        [Required, MaxLength(25), MinLength(5)]
        public string KnownAs { get; set; }
        public string Add { get; set; }
        public string Add2 { get; set; }
        [Required, MaxLength(50), MinLength(4)]
        public string City { get; set; }
        public string Pin { get; set; }
        public string District { get; set; }
        public string State { get; set; }
        public string Country { get; set; }="India";
        [Required, EmailAddress]
        public string Email {get; set;}
        public string Website {get; set;}
        public string Phone {get; set;}
        public string Phone2 {get; set;}
        public DateTime CreatedOn { get; set; }= DateTime.UtcNow;
        public string Introduction { get; set; }
        public string CustomerStatus {get; set;}="Active";
        public bool IsBlackListed { get; set; } = false;
        public ICollection<CustomerIndustry> CustomerIndustries { get; set; }
        public ICollection<CustomerOfficial> CustomerOfficials { get; set; }
        public virtual ICollection<AgencySpecialty> AgencySpecialties {get; set;}
       
    }
}